Fixed NPE in getGLobalQuantity for Development Build Fail

PhilMFischer opened the pull request:
- Directly accessing the bean for the unitQuantity and checkign if it is null.
- Changes to the Code Generators have to be handled in CORE. See: https://github.com/virtualsatellite/VirtualSatellite4-Core/issues/547

---
Closes #99: Development Build is crashing by NPE

Commit:
Fixed NPE in getGLobalQuantity - (Task #99)

- Directly accessing the bean for the unitQuantity and checkign if it is
null.

---
Task #99: Development Build is crashing by NPE

Commit:
Fixing other places where quantity value is accessed - (Task #99)

- Fixing INterfaceEnd quantity.
- Fixing decorator for IFE quantities.

---
Task #99: Development Build is crashing by NPE

andriikovalov-dlr commented on de.dlr.sc.virsat.model.extension.cef.interfaces/src/de/dlr/sc/virsat/model/extension/cef/interfaces/model/AInterfaceEnd.java:
```diff
@@ -63,11 +63,12 @@ public AInterfaceEnd(CategoryAssignment categoryAssignment) {
-			globalUnitQuantity *= 1;
-		}
+
+		// Workaround due to API Inconsistency with Beans see https://github.com/virtualsatellite/VirtualSatellite4-CEF/issues/99
```
Let's maybe do it something like
`globalUnitQuantity *= isSetQuantity() ? getQuantity() : 1;`

andriikovalov-dlr commented on de.dlr.sc.virsat.model.extension.cef/src/de/dlr/sc/virsat/model/extension/cef/model/EquipmentParameters.java:
```diff
@@ -61,11 +61,12 @@ public EquipmentParameters(CategoryAssignment categoryAssignment) {
-		}
+		
+		// Workaround due to API Inconsistency with Beans see https://github.com/virtualsatellite/VirtualSatellite4-CEF/issues/99
+		Long unitQauntity = getUnitQuantityBean().getValue();
```
Let's maybe do it something like
`globalUnitQuantity *= isSetQuantity() ? getQuantity() : 1;`
Then it is not a temporary workaround anymore :)

andriikovalov-dlr commented on de.dlr.sc.virsat.model.extension.cef.interfaces.ui/src/de/dlr/sc/virsat/model/extension/cef/interfaces/ui/decorator/InterfaceEndQuantityKindDecorator.java:
```diff
@@ -42,10 +42,10 @@ public void decorate(Object element, IDecoration decoration) {
 				} else if ((iBca != null) && (iBca instanceof AInterfaceEnd)) {
 					AInterfaceEnd aIfe = (AInterfaceEnd) iBca;
-					decorateQuantity(decoration, aIfe.getQuantity());
+					decorateQuantity(decoration, aIfe.getQuantityBean().getValue());
```
I think it doesn't save us from an NPE, since `decorateQuantityAndType` wants a primitive long. Let's maybe change argument type to `Long`, and check for null in `decorateQuantityAndType`?

Commit:
Adjustments due to review - (Task #99)

- Focus on is set method rather than null checks upfront accessing the
long value.

---
Task #99: Development Build is crashing by NPE

andriikovalov-dlr committed:
Simplified checks for get global unit quantity - (Task #99)

---
Task #99: Development Build is crashing by NPE

[skip ci]

andriikovalov-dlr committed:
Refactored quantity decorators - (Task #99)

---
Task #99: Development Build is crashing by NPE
